### ISSUE TITLE
App links against libcrypto.so which is not in NDK!

### ISSUE THREAD

**Issue by kruton** [aluvalasuman/ics-openvpn-eclipseversion#1]
The library `libopvpnutil.so` links against an ***ABI unstable*** `libcrypto.so` from some random version of the platform. This should either be statically linked with the crypto library or a differently-named version should be included (e.g., don't use the filename `libcrypto.so` because it conflicts with the platform).

**Comment by aluvalasuman** [comment]
Thanks @kruton. This copy was created for the people who are not comfortable with Studio. But your suggestion is valid. I may not be updating this. So, people who will clone or fork please fix the @kruton comment before use.

**Comment by petergjx** [comment]
@aluvalasuman hello aluvalasuman, now i use your project to build apk run on android5.0, here cames error info as "Error: only position independent executables (PIE) are supported", i also add "LOCAL_CFLAGS += -fPIE LOCAL_LDFLAGS += -fPIE -pie" to "BUILD_EXECUTABLE" and "LOCAL_CFLAGS += -fPIC LOCAL_LDFLAGS += -fPIE -pie" to "BUILD_SHARED_LIBRARY", but it is still not work, so could you give suggestion to me for this?thanks.

**Comment by aluvalasuman** [comment]
@petergjx , you can change the platform version in the jni/Application.mk file "APP_PLATFORM := android-14" to "APP_PLATFORM := android-16" and try.

**Comment by JiguangXiao** [comment]
@aluvalasuman yes, i had try this but get the same error, then i use those flags for compilation, the openvpn is still with no pie on it.

**Comment by aluvalasuman** [comment]
@JiguangXiao , Don't have idea.

**Comment by JiguangXiao** [comment]
@aluvalasuman, i use android studio instead, it works well.
